plugin/storage/es/spanstore/reader.go

```diff
@@ -234,9 +234,35 @@ func (s *SpanReader) FindTraces(ctx context.Context, traceQuery *spanstore.Trace
 	return s.multiRead(ctx, uniqueTraceIDs, traceQuery.StartTimeMin, traceQuery.StartTimeMax)
 }
 
-// FindTraceIDs is not implemented.
+// FindTraceIDs retrieves traces IDs that match the traceQuery
 func (s *SpanReader) FindTraceIDs(ctx context.Context, traceQuery *spanstore.TraceQueryParameters) ([]model.TraceID, error) {
-	return nil, errors.New("not implemented") // TODO: Implement
+	if err := validateQuery(traceQuery); err != nil {
```

Comment:
I see a lot of repeated code between `FindTraces` and `FindTraceIDs`, are we able to refactor such that the initial validation and retrieval are shared?

Also, could you create a new span here - similar to L221

Reply:
Ok, thanks.